consensus/src/dag/dag_driver.rs

```diff
@@ -47,20 +47,75 @@ pub enum DagDriverError {
     MissingParents,
 }
 
+/// Interface to decide if we should move forward to a new round
+#[async_trait]
+pub trait RoundState: Send {
+    async fn update(
+        &mut self,
+        highest_round: Round,
+        strong_links_for_highest_round: Option<Vec<NodeCertificate>>,
+    );
+    fn enter_new_round(&mut self, new_round: Round);
+
+    fn current_round(&self) -> Round;
+}
+
+pub struct OptimisticResponsiveRound {
+    current_round: Round,
+    event_sender: tokio::sync::mpsc::Sender<Round>,
+}
+
+impl OptimisticResponsiveRound {
+    pub fn new(event_sender: tokio::sync::mpsc::Sender<Round>) -> OptimisticResponsiveRound {
+        OptimisticResponsiveRound {
+            current_round: 0,
+            event_sender,
+        }
+    }
+}
+
+#[async_trait]
+impl RoundState for OptimisticResponsiveRound {
+    async fn update(
```

Comment:
nit: no "update" actually happens here? 

this should be `enter_new_round` right? the below `enter_new_round` function should just be a setter like `set_new_round`?

Reply:
I think this is more `check_for_new_round`, it doesn't necessarily enter